packages/eas-cli/src/credentials/ios/appstore/authenticate.ts

```diff
@@ -144,6 +178,40 @@ async function loginWithUserCredentialsAsync({
 }
 
 export async function authenticateAsync(options: Options = {}): Promise<AuthCtx> {
+  if (options.mode === AuthenticationMode.API_KEY) {
+    return await authenticateWithApiKeyAsync(options);
+  } else {
+    return await authenticateAsUserAsync(options);
+  }
+}
+
+// TODO: this may cause undefined behaviour in third party code
+async function authenticateWithApiKeyAsync(options: Options = {}): Promise<ApiKeyAuthCtx> {
+  const { ascApiKey, teamId, teamName, teamType } = options;
+  assert(ascApiKey, 'ascApiKey must be defined');
+  assert(teamId && teamType !== undefined, 'teamId and teamType must be defined');
+  const isInHouse = teamType === 'inHouse';
+  return {
```

Comment:
@EvanBacon this object is missing a lot of expected fields like  username, cookies, session. Everything works fine when i test it out manually, but it may lead to some unexpected behaviour down the road if third party is expecting those fields to be present

Reply:
Makes sense, originally we moved a lot of the authentication handling into EAS CLI so it was clearer to users how we handled/stored their credentials.